include/gz/math/Inertial.hh

```diff
@@ -112,6 +133,28 @@ namespace gz
         return this->pose;
       }
 
+      /// \brief Set the matrix representing the inertia of the fluid that is
+      /// dislocated when the body moves. Added mass should be zero if the
+      /// density of the surrounding fluid is negligible with respect to the
+      /// body's density.
+      ///
+      /// \param[in] _m New Matrix6 object, which must be a symmetric matrix.
+      /// \return True if the Matrix6 is symmetric.
+      /// \sa SpatialMatrix
+      public: bool SetFluidAddedMass(const Matrix6<T> &_m)
+      {
+        this->addedMass = _m;
+        return this->addedMass.value() == this->addedMass.value().Transposed();
+      }
+
+      /// \brief Get the fluid added mass matrix.
+      /// \return The added mass matrix. It will be nullpot of the added mass
+      /// was never set.
+      public: std::optional< Matrix6<T> > FluidAddedMass() const
```

Comment:
return a `const` reference?

Reply:
Done in [de61ec1](https://github.com/gazebosim/gz-math/pull/459/commits/de61ec19b697f2c9b97febb36cf0b460c3edd986). I haven't found any example code doing this though, but I think it should provide the functionality of `std::optional` while preventing a copy.